File: src/main/templates/python_sdk/src/api/page_result.py
"""
分页结果类
"""

from typing import List, TypeVar, Generic, Dict, Any

T = TypeVar('T')


class PageResult(Generic[T]):
    """分页结果类"""
    
    def __init__(self, total: int = 0, items: List[T] = None):
        """
        初始化分页结果
        
        Args:
            total: 总数
            items: 数据列表
        """
        self.total = total
        self.items = items or []
    
    def get_total(self) -> int:
        """获取总数"""
        return self.total
    
    def set_total(self, total: int) -> None:
        """设置总数"""
        self.total = total
    
    def get_items(self) -> List[T]:
        """获取数据列表"""
        return self.items
    
    def set_items(self, items: List[T]) -> None:
        """设置数据列表"""
        self.items = items
    
    def get_size(self) -> int:
        """获取当前页数据数量"""
        return len(self.items)
    
    def is_empty(self) -> bool:
        """是否为空"""
        return len(self.items) == 0
    
    def to_dict(self) -> Dict[str, Any]:
        """转换为字典"""
        return {
            'total': self.total,
            'items': self.items
        }
    
    def __repr__(self) -> str:
        return f"PageResult(total={self.total}, size={self.get_size()})"

File: src/main/templates/javascript_sdk/src/api/PageResult.js
/**
 * 分页结果类
 */
export class PageResult {
  constructor(total = 0, list = []) {
    this.total = total;
    this.list = list;
  }

  /**
   * 获取总数
   */
  getTotal() {
    return this.total;
  }

  /**
   * 设置总数
   */
  setTotal(total) {
    this.total = total;
  }

  /**
   * 获取列表
   */
  getList() {
    return this.list;
  }

  /**
   * 设置列表
   */
  setList(list) {
    this.list = list;
  }

  /**
   * 获取当前页数据数量
   */
  getSize() {
    return this.list.length;
  }

  /**
   * 是否为空
   */
  isEmpty() {
    return this.list.length === 0;
  }

  /**
   * 转换为JSON
   */
  toJSON() {
    return {
      total: this.total,
      list: this.list
    };
  }
}
